### PR TITLE
Add `Etc` module with methods for fetching user information

### PR DESCRIPTION
**Pull request by tmatilai**
¡Hola!

Would you like to include [Ruby inspired](http://ruby-doc.org/stdlib-2.3.0/libdoc/etc/rdoc/Etc.html) `Etc` module to the standard library? Or do I publish it as shard?

Disclaimer: I just started playing with the cool new language today, so might have some funny things in it.

### CODE REVIEW

File: src/etc.cr [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  struct Passwd
+    property name
+    property passwd
+    property uid
+    property gid
+    property gecos
+    property dir
+    property shell
+
+    def initialize(pwd : LibC::Passwd)
```
Comment (bcardiff) [inline src/etc.cr]: I would say that LibC::Passwd should not appear here. The wrapper module should perform the conversion IMO.

Also, take a look at http://crystal-lang.org/api/toplevel.html#record%28name%2C%2Aproperties%29-macro you will probably like it.

Comment (tmatilai) [inline src/etc.cr]: Agreed. Naming is hard, but I pushed a new commit trying to fix that. (As a separate commit for clarity, will squash it later.)

File: src/etc.cr [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      gecos : Char*
+      dir : Char*
+      shell : Char*
+    end
+  end
+
+  fun getlogin : Char*
+  fun getpwnam(name : Char*) : Passwd*
+  fun getpwuid(uid : UInt32) : Passwd*
+  fun getuid : UInt32
```
Comment (tmatilai) [inline src/etc.cr]: `getuid` should propably be defined somewhere else? In `process.cs`?

**Comment by asterite** [comment]
Hola!

Thank you for this. We have to analyze this a bit. It's way too unix specific. What would happen in Windows, this module won't be available? I think the standard library shouldn't have too platform-specific functionality, but I don't know.

**Comment by tmatilai** [comment]
Seems that in Ruby the non-existing methods just return `nil`. And `getlogin` on Windows returns value of  `USER` or `USERNAME` environment variable. And anyway, in Ruby the module has a couple of other methods too.

Your call, for me this was just a fun exercise. =)

**Comment by jhass** [comment]
If we want this it should go to the new `System` module and avoid prefixes such as `get`.

**Comment by asterite** [comment]
Yes, having this in `System` sounds good. I'd check with @ysbaddaden to generate the correct bindings, though.

**Comment by ysbaddaden** [comment]
Bindings for `getlogin` and related syscalls should already be available in [posix](https://github.com/ysbaddaden/posix), they just need to be enabled for Crystal, then build targets again.

**Comment by miketheman** [comment]
#codetriage 
So it appears that this was left in a state where the following seem to be the path forward:

- enable bindings for relevant syscalls by submitting & merging a PR to https://github.com/ysbaddaden/posix/
- Do something with the change (uncertain of what this is)
- Add some of the code & specs from this PR under the `System` module without `get` prefix, so `System.login` would return `miketheman` on my system
- Profit!!

If that's correct and desirable, I'm happy to take a stab at this, but I'm uncertain of what Step 2 would be.
@ysbaddaden the README on the posix repo isn't very clear as to what happens next with changes there.

**Comment by straight-shoota** [comment]
It seems like this PR is mostly superseeded by the more recent #5627 and #4800

**Pull request closed without merging**